src/client/dfuse/pil4dfs/hook.c

```diff
@@ -914,14 +918,14 @@ register_a_hook(const char *module_name, const char *func_name, const void *new_
 		get_module_maps_inited = 1;
 	}
 
-	if (strcmp(module_name, "ld") == 0)
-		strcpy(module_name_local, path_ld);
-	else if (strcmp(module_name, "libc") == 0)
-		strcpy(module_name_local, path_libc);
-	else if (strcmp(module_name, "libpthread") == 0)
-		strcpy(module_name_local, path_libpthread);
+	if (strncmp(module_name, "ld", 3) == 0)
```

Comment:
i think strncmp should not include the NULL terminator, so it should be in this case strlen("ld") instead of 3.
strncpy should be strlen("ld")+1

same comment applies to the others.

Reply:
The NULL terminator is included on purpose, so exact full string matching is used. It is not only matching the initial sub-string.